### PR TITLE
add miniver, get __version__ from git tags

### PR DESCRIPTION
**Pull request by basnijholt**
Currently, there is no way of finding out what version people have installed.

This makes sure that `qms.__version__` is always correctly defined, using [semver](https://semver.org/).

We would for example tag `1.1.0` now by doing:
```
git tag v1.1.0 -m "version 1.1.0"
```

It yields version strings like `1.1.0`, `1.1.0+2.g1076c97`, or `0.11+2.g1076c97.dirty`.

Where `1.1.0+2.g1076c97` means, 2 commits after the tag `1.1.0` and on the git hash 1076c97,
`dirty` means that there are unstaged files.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,9 +4,30 @@
 
 from setuptools import setup
 
+
+if sys.version_info < (3, 6):
```
Comment (basnijholt) [inline setup.py]: This will execute too if users have `pip < 9.0.1`, which didn't know about `setup(..., python_requires='...')`.

File: setup.py [hunk 2]
```diff
@@ -4,9 +4,30 @@
...
+    return module.__version__, module.cmdclass
+
+
+version, cmdclass = get_version_and_cmdclass("qmt")
+
 setup(
     name="qmt",
-    version="1.0",
+    version=version,
+    python_requires=">=3.6",
```
Comment (basnijholt) [inline setup.py]: `python_requires` will tell to pip ahead of time not to even try to install on incompatible versions

File: qmt/__init__.py [hunk 3]
```diff
@@ -1,4 +1,9 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
+
+from ._version import __version__
+
+del _version
```
Comment (likewei92) [inline qmt/__init__.py]: Sorry what does this del do? Does `from ._version import __version__` create a `_version` variable?

File: qmt/_version.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+# This file is part of 'miniver': https://github.com/jbweston/miniver
```
Comment (likewei92) [inline qmt/_version.py]: Are these files a result of setting up miniver? How would we upgrade (if we need to)?

Comment (basnijholt) [reply qmt/_version.py]: Yes, this is required to get the version from the git tags.

We upgrade by just downloading the new file from that repo.

File: qmt/__init__.py (hunk 3 above)
Comment (basnijholt) [reply qmt/__init__.py]: Exactly what you say. It's just to not confuse the users.

**Pull request merged**